File: Definitions/Game_Definitions/classes/DestructibleMasterDevice.lua
---@meta
---@diagnostic disable

---@class DestructibleMasterDevice : InteractiveMasterDevice
DestructibleMasterDevice = {}

---@return DestructibleMasterDevice
function DestructibleMasterDevice.new() return end

---@param props table
---@return DestructibleMasterDevice
function DestructibleMasterDevice.new(props) return end

---@return Bool
function DestructibleMasterDevice:OnDetach() return end

---@param evt entPhysicalDestructionEvent
---@return Bool
function DestructibleMasterDevice:OnPhysicalDestructionEvent(evt) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function DestructibleMasterDevice:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function DestructibleMasterDevice:OnTakeControl(ri) return end

---@return DestructibleMasterDeviceController
function DestructibleMasterDevice:GetController() return end

---@return DestructibleMasterDeviceControllerPS
function DestructibleMasterDevice:GetDevicePS() return end

File: DecompiledGameScripts/cyberpunk/devices/destructible/destructibleMasterDevice.swift

public class DestructibleMasterDevice extends InteractiveMasterDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as DestructibleMasterDeviceController;
  }

  protected cb func OnDetach() -> Bool {
    super.OnDetach();
  }

  protected cb func OnPhysicalDestructionEvent(evt: ref<PhysicalDestructionEvent>) -> Bool {
    this.GetDevicePS().RefreshSlaves_Event();
  }

  protected const func GetController() -> ref<DestructibleMasterDeviceController> {
    return this.m_controller as DestructibleMasterDeviceController;
  }

  public const func GetDevicePS() -> ref<DestructibleMasterDeviceControllerPS> {
    return this.GetController().GetPS();
  }
}
